src/WebJobs.Script/Diagnostics/FileTraceWriter.cs

```diff
@@ -100,30 +83,45 @@ public override void Flush()
                 sb.AppendLine(line);
             }
 
+            if (_currentLogFileInfo == null)
+            {
+                // delay create log file
+                SetLogFile();
```

Comment:
Not pushing back, but curious -- delaying the file creation isn't directly related to the issue, is it? The core of the fix seems to be the empty file creation, which we technically could have done in the constructor.

Reply:
Yeah, initially I had this in the constructor, but realized on startup that it would mean eager creation of a file for each function, whether or not that function is active. So I made the further improvement to move it here. But yes - the actual fix is to create the file under a lock, so multiple instances racing to create a file will converge on one.